corgi/tasks/brew.py

```diff
@@ -576,35 +584,25 @@ def fetch_unprocessed_relations(
         query &= Q(created_at__gte=created_since)
     relations_query = (
         ProductComponentRelation.objects.filter(query)
-        .values_list("build_id", "build_type")
-        .distinct()
+        .filter(software_build=None)
         .using("read_only")
     )
 
     processed_builds = 0
-    for build_id, build_type in relations_query.iterator():
-        if not build_id:
-            # build_id defaults to "" and int() will fail in this case
-            continue
-        if (
-            save_only
-            or not SoftwareBuild.objects.filter(build_id=build_id, build_type=build_type)
-            .using("read_only")
-            .exists()
-        ):
-            logger.info("Processing %s relation build with id: %s", relation_type, build_id)
-            if build_type == SoftwareBuild.Type.CENTOS:
-                # This skips use of the Collector models for builds in the CENTOS koji instance
-                # It was done to avoid updating the collector models not to use build_id as
-                # a primary key. It's possible because the only product stream (openstack-rdo)
-                # stored in CENTOS koji doesn't use modules
-                slow_fetch_brew_build.delay(
-                    build_id, SoftwareBuild.Type.CENTOS, force_process=force_process
-                )
-                processed_builds += 1
-                continue
-            slow_fetch_modular_build.delay(build_id, force_process=force_process)
+    for relation in relations_query.iterator():
+        logger.info(f"Processing {relation.type} relation build with id: {relation.build_id}")
+        if relation.build_type == SoftwareBuild.Type.CENTOS:
+            # This skips use of the Collector models for builds in the CENTOS koji instance
+            # It was done to avoid updating the collector models not to use build_id as
+            # a primary key. It's possible because the only product stream (openstack-rdo)
+            # stored in CENTOS koji doesn't use modules
+            slow_fetch_brew_build.delay(
+                relation.build_id, SoftwareBuild.Type.CENTOS, force_process=force_process
+            )
             processed_builds += 1
+            continue
+        slow_fetch_modular_build.delay(relation.build_id, force_process=force_process)
+        processed_builds += 1
```

Comment:
Minor: I think if / else is a little clearer than if / continue, since it avoids duplicating the processed_builds counter:
```suggestion
        else:
            slow_fetch_modular_build.delay(relation.build_id, force_process=force_process)
        processed_builds += 1
```

Reply:
good idea.